cluster-controller/src/main/java/io/strimzi/controller/cluster/ClusterControllerConfig.java

```diff
@@ -3,22 +3,60 @@
 import java.util.HashMap;
 import java.util.Map;
 
+/**
+ * Cluster Controller configuration
+ */
 public class ClusterControllerConfig {
 
     public static final String STRIMZI_NAMESPACE = "STRIMZI_NAMESPACE";
     public static final String STRIMZI_CONFIGMAP_LABELS = "STRIMZI_CONFIGMAP_LABELS";
+    public static final String STRIMZI_FULL_RECONCILIATION_INTERVAL = "STRIMZI_FULL_RECONCILIATION_INTERVAL";
+
+    private static final long DEFAULT_FULL_RECONCILIATION_INTERVAL = 120_000; // in ms (2 minutes)
 
     private Map<String, String> labels;
     private String namespace;
+    private long reconciliationInterval;
 
-    public ClusterControllerConfig(String namespace, Map<String, String> labels) {
+    /**
+     * Constructor
+     *
+     * @param namespace namespace in which the controller will run and create resources
+     * @param labels    labels used for watching the cluster ConfigMap
+     * @param reconciliationInterval    specify every how many milliseconds the reconciliation runs
+     */
+    public ClusterControllerConfig(String namespace, Map<String, String> labels, long reconciliationInterval) {
         this.namespace = namespace;
         this.labels = labels;
+        this.reconciliationInterval = reconciliationInterval;
     }
 
-    public static ClusterControllerConfig fromEnv() {
-        String namespace = System.getenv(ClusterControllerConfig.STRIMZI_NAMESPACE);
-        String stringLabels = System.getenv(ClusterControllerConfig.STRIMZI_CONFIGMAP_LABELS);
+    /**
+     * Constructor which provide a configuration with a default (120000 ms) reconciliation interval
```

Comment:
Is the overload really necessary? It's not like there are any clients of this API, so we could fix the call sites. There can only be a couple.